### PR TITLE
rewrite accessibility section on invokers

### PR DESCRIPTION
**Pull request by keithamus**
This rewrites the a11y section on invokers to be far more comprehensive, detailing the exact intentions we have when it comes to applying various ARIA states to the button, as well as focus fixups.

### CODE REVIEW

**Review by mfreed7 — APPROVED** [review]
Wow, awesome new section! This looks good to me, but I'm not the right person to judge it. I'll stamp this approved, because I'm ok landing it and putting up PRs to improve it over time. But @scottaohara and @aleventhal please take a look and also @ mention anyone else that should have a look.

**Comment by scottaohara** [comment]
i'll review this later today or tomorrow.  thanks @keithamus

File: site/src/pages/components/invokers.explainer.mdx
Comment (scottaohara) [inline site/src/pages/components/invokers.explainer.mdx:429]: this isn't really accurate - re: parent/child relationship.  i'm not sure why this is mentioned at all if it's not being used / was never implemented for popovers?

File: site/src/pages/components/invokers.explainer.mdx
Comment (scottaohara) [inline site/src/pages/components/invokers.explainer.mdx:439]: same comments as before.  the intent for the haspopup was more to help users know what types of popups were invoked from a combobox.  i really am trying to get ARIA back on track for that being the primary purpose of this attribute, i really dont want to see this get used for buttons and further cement a practice this wasn't originally intended for.

File: site/src/pages/components/invokers.explainer.mdx
Comment (scottaohara) [inline site/src/pages/components/invokers.explainer.mdx:495]: it's weird this says aria-expanded would be the correct state to expose, but in the next section also talks about how the state is redundant.

File: site/src/pages/components/invokers.explainer.mdx
Comment (scottaohara) [inline site/src/pages/components/invokers.explainer.mdx:509]: similar as before, the description here is not accurate, but also just doesn't seem worth mentioning if the only reason to do so is to say "it's not used"

**Review by scottaohara — COMMENTED** [review]
let me know if there are questions with my comments

**Comment by mfreed7** [comment]
> let me know if there are questions with my comments


@keithamus is the plan still to update this with @scottaohara 's comments before landing? (I hope so.)

**Comment by scottaohara** [comment]
we are going to need the HTML AAM PR for this as well, so whether fixed here or in that PR (and then this gets updated to just say 'go look at html aam' instead) is fine by me.  

but from what i understand, we are going to need that HTML AAM PR to land at the same / close to the same time as the HTML PR.  so we need to get that going / address the above either way.

File: site/src/pages/components/invokers.explainer.mdx
Comment (keithamus) [reply site/src/pages/components/invokers.explainer.mdx:429]: I tried to add these extra bits as somewhat of an FAQ to field questions of "why hasn't this particular property been implemented", to have some written down rationale that people can refer to when those questions arise.

Comment (scottaohara) [reply site/src/pages/components/invokers.explainer.mdx:429]: ok, that makes sense.  

so then it's more of a way this is described then - since it's not commonly a parent/child relationship - but more of an association between related elements

Comment (keithamus) [reply site/src/pages/components/invokers.explainer.mdx:429]: Would it be better to point out instead that `aria-details` has better support, and applying both is redundant?

Comment (scottaohara) [reply site/src/pages/components/invokers.explainer.mdx:429]: yah that could work.  so something like

aria-controls is not used due to lack of support among screen readers, and even if it was supported it would be redundant to the association already created due to the implicit aria-details relationship already created.

**Comment by keithamus** [comment]
@scottaohara I believe I've addressed your review commentary. If you would kindly give it another pass that would be much appreciated.

File: site/src/pages/components/invokers.explainer.mdx
Comment (scottaohara) [inline site/src/pages/components/invokers.explainer.mdx:458]: wording nit to not be redundant in saying it 'is not useful' and then 'it would be useless...."

```suggestion
is closed, and non-navigable for a dialog that is open as modal, so it would 
provide no benefit to add.
```

File: site/src/pages/components/invokers.explainer.mdx
Comment (scottaohara) [inline site/src/pages/components/invokers.explainer.mdx:476]:
```suggestion
When the `commandfor=` element is a `<dialog>`, `aria-haspopup=dialog` might seem
a potentially viable attribute to add. Some Assistive Technologies will announce
```

File: site/src/pages/components/invokers.explainer.mdx
Comment (scottaohara) [inline site/src/pages/components/invokers.explainer.mdx:481]:
```suggestion
rarely come with additional visual treatment, and so the [lack of a visual
analog means that using this attribute likely provides surplus information](https://w3c.github.io/aria/#h-note-50)
```

**Review by scottaohara — APPROVED** [review]
approving with minor additional suggestions.  take them as you will.  thank you!

**Comment by keithamus** [comment]
Thanks @scottaohara! I'll merge and get to work on the AAM PRs!

**Comment by scottaohara** [comment]
> Thanks @scottaohara! I'll merge and get to work on the AAM PRs!

excellent. thank you.  just give me a holler if you have any questions when making that PR.

**Pull request merged**
